eql/ast.py

```diff
@@ -944,6 +944,32 @@ def _render(self):
         return text
 
 
+class Sample(EqlNode):
+    """Sample finds events matching the defined filters, regardless of their temporal order.
+
+    Sample supports defining one or more join keys.
+    """
+
+    __slots__ = 'queries', 'join_keys'
+
+    def __init__(self, queries, join_keys=None):
+        """Create a Sample of multiple events.
+
+        :param list[SubqueryBy] queries: List of queries to be sampled
+        :param list[str] join_keys: List of join keys
+        """
+        self.queries = queries
+        self.join_keys = join_keys or []
+
+    def _render(self):
+        text = 'sample'
+        if self.join_keys:
+            text += ' by ' + ', '.join(self.join_keys)
+        text += '\n'
+        text += self.indent('\n'.join(query.render() for query in self.queries))
```

Comment:
Calling query.render() adds the sample's join keys to each subquery. Not sure if this is correct.

Reply:
https://github.com/endgameinc/eql/pull/74/commits/711fb45dde009bee4cfcb1ce66233409e7b7b057